container.fc

```diff
@@ -162,3 +162,7 @@ HOME_DIR/\.local/share/containers/storage/volumes/[^/]*/.*	gen_context(system_u:
 /var/log/lxc(/.*)?		gen_context(system_u:object_r:container_log_t,s0)
 /var/log/lxd(/.*)?		gen_context(system_u:object_r:container_log_t,s0)
 /etc/kubernetes(/.*)?		gen_context(system_u:object_r:kubernetes_file_t,s0)
+
+/opt/kata/bin(/.*)?		gen_context(system_u:object_r:container_runtime_exec_t,s0)
```

Comment:
This is the location of a bunch of binaries. FWIW some of these have different labels on Fedora :

```
system_u:object_r:container_runtime_exec_t:s0 /usr/bin/containerd-shim-kata-v2
                   system_u:object_r:bin_t:s0 /usr/bin/kata-collect-data.sh
                   system_u:object_r:bin_t:s0 /usr/bin/kata-monitor
                   system_u:object_r:bin_t:s0 /usr/bin/kata-runtime
             system_u:object_r:qemu_exec_t:s0 /usr/bin/qemu-system-x86_64
                   system_u:object_r:bin_t:s0 /usr/libexec/kata-containers/agent/usr/bin/kata-agent
```

as well as in RHEL :

```
system_u:object_r:container_runtime_exec_t:s0 /usr/bin/containerd-shim-kata-v2
                   system_u:object_r:bin_t:s0 /usr/bin/kata-collect-data.sh
                   system_u:object_r:bin_t:s0 /usr/bin/kata-log-parser
                   system_u:object_r:bin_t:s0 /usr/bin/kata-monitor
                   system_u:object_r:bin_t:s0 /usr/bin/kata-runtime
                   system_u:object_r:bin_t:s0 /usr/libexec/kata-containers/agent/usr/bin/kata-agent
             system_u:object_r:qemu_exec_t:s0 /usr/libexec/qemu-kvm
```

I don't know about the other executables.

Reply:
TBH I wasn't sure how to label them. Yes, on stock fedora they are mainly `bin_t` but since all files within this `container.fc` were using `container_runtime_exec_t` I used that one instead (after verifying it works on `centos8stream`).

Reply:
Are these container engines?  Or is this something launched by containerd/cri-o?

Reply:
ps -eZ | grep DAEMONS 

To tell me what they  run with currently?

Reply:
I don't know the details, perhaps @gkurz would know more. There are no `DAEMONS` processes running on the system, but according the `pstree` the processes are spawned from `containerd-shim`:

```
        ├─containerd-shim─┬─qemu-system-x86───3*[{qemu-system-x86}]
        │                 ├─virtiofsd───virtiofsd2───3*[{virtiofsd2}]
        │                 └─10*[{containerd-shim}]
```